### DISCUSSION TITLE
CAN Transport layer

### DISCUSSION THREAD

**Discussion by vChavezB** [intercreate/smpmgr#17] [General]
First of all thanks for this project, it looks like a promising alternative to the rust and go variants :)

I just wanted to give an idea. I am working at the moment with some automotive mcus from NXP with Zephyr. The devices I am building only have a CAN interface. For this reason in the near future I might need to add a transport layer to MCU Boot with CAN.  

I like the smpclient project as it has an abstraction to add more protocols, so thanks for that!

My proposal is to add a CAN interface. The project from https://github.com/hardbyte/python-can might be a good solution to abstract CAN with python.

If we decide to add CAN support we will probably use this project as a starting point. 👍

**Comment by JPHutchins** [comment]
Adding this discussion for context since CAN was validated to some extent: https://github.com/intercreate/smpclient/issues/19

MCUBoot fork/commit with CAN as SMP transport: https://github.com/vChavezB/mcuboot/commit/60872b4dbfe9c3c613991051d2463f16803ed27e
